lang/syn/src/codegen/program/entry.rs

```diff
@@ -63,8 +63,15 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
 
             dispatch(program_id, accounts, data)
                 .map_err(|e| {
-                    anchor_lang::solana_program::msg!(&e.to_string());
-                    e
+                    let program_error = e.to_program_error();
+                    anchor_lang::solana_program::msg!(&program_error.to_string());
+                    if let ::anchor_lang::solana_program::program_error::ProgramError::Custom(code) = program_error {
```

Comment:
This printing is late-night hackyness, it seems that the anchor ts code isn't super lenient with error parsing, so I'm just printing both the old `Custom error code: 0x` thing as well as the new one.